### PR TITLE
Run a daily job to flag reports from repeat writers

### PR DESCRIPTION
**Pull request by eileen-nava**
[Exact Description Match - Write a job that identifies reports associated with repeat writers.](https://github.com/usdoj-crt/crt-portal-management/issues/1291)

## What does this change?
This PR creates a management command that will run once a day in all environments.  The management command will set the `cts_forms_report` boolean field `by_repeat_writer` to true for:
A) reports that have the exact same violation summary as at least 50** other reports
B) reports whose contact email is associated with a 'CRT - Constant Writer' email
C) reports whose contact email is associated with a report that caused an intake specialist to print the 'CRT - Constant Writer' template

** NOTE: During stand-up @blaughman said that he spoke with other agencies that receive repeat writers, and there isn't a formally agreed upon number of reports at which someone is considered a repeat writer.  I think it's still an open question if 50 is the correct cut-off.  I can easily change this limit after consulting with @blaughman and @randyabramson, but for now I have set it to 50.

## Screenshots (for front-end PR):
n/a (This isn't a front-end PR)

## Checklist:
+ [x] Confirm all tests pass
+ [x] Create 75 reports that have the exact same violation summary.  Manually run the management command, then confirm that `by_repeat_writer` equals `True` for all 75 reports with duplicate violation summaries.
+ [ ] Create a report, send the "constant writer" email in response to it, and then create another report with the same contact email as the first report.   Manually run the management command, then confirm that `by_repeat_writer` equals `True` for the second report.
+ [ ] Create a report, print the "constant writer" template in response to it, and then create another report with the same contact email as the first report.   Manually run the management command, then confirm that `by_repeat_writer` equals `True` for the second report.

### Author

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [ ] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/management/commands/flag_repeat_writers.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+class Command(BaseCommand):
+    help = "Flag reports that were submitted by repeat writers."
+
+    def handle(self, *args, **options):
+        repeat_summary_sql = """
+        WITH repeat_summary AS (SELECT violation_summary, COUNT(*)
+        FROM cts_forms_report
+        GROUP BY violation_summary
+        HAVING count(*) > 50 )
```
Comment (JackRyan1989) [inline crt_portal/cts_forms/management/commands/flag_repeat_writers.py:13]: Do we want this to be greater than or equal to 50? Right now won't it only pull in reports with a count of 51 and higher?

File: crt_portal/cts_forms/management/commands/flag_repeat_writers.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        UPDATE cts_forms_report
+        SET by_repeat_writer = true
+        WHERE id IN
+        (SELECT report.id
+        FROM cts_forms_report report
+        JOIN repeat_summary
+        ON report.violation_summary = repeat_summary.violation_summary
+        ORDER BY report.violation_summary);
+        """
```
Comment (JackRyan1989) [inline crt_portal/cts_forms/management/commands/flag_repeat_writers.py:22]: This is cool! I think I'm gonna need you to walk me through this query though...

File: crt_portal/cts_forms/management/commands/flag_repeat_writers.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        cursor.execute(repeat_summary_sql)
+
+        # Flag reports that were submitted by individuals whose contact email addresses had already received a repeat writer email
+        repeat_writer_emailed_sql = """
+        UPDATE cts_forms_report
+        SET by_repeat_writer = true
+        WHERE contact_email IN (
+        SELECT array_to_string(regexp_matches(description, '\\w+@\\S+', 'g'), ';') FROM actstream_action act
+        WHERE act.verb = 'Contacted complainant:'
+        AND act.description LIKE '%Constant Writer%');
```
Comment (JackRyan1989) [inline crt_portal/cts_forms/management/commands/flag_repeat_writers.py:34]: I need help understanding this

File: crt_portal/cts_forms/management/commands/flag_repeat_writers.py (hunk 1 above)
Comment (eileen-nava) [reply crt_portal/cts_forms/management/commands/flag_repeat_writers.py:13]: Good point.  I will change that.

File: crt_portal/cts_forms/management/commands/flag_repeat_writers.py (hunk 2 above)
Comment (eileen-nava) [reply crt_portal/cts_forms/management/commands/flag_repeat_writers.py:22]: Sure thing 👍🏻

Comment (eileen-nava) [reply crt_portal/cts_forms/management/commands/flag_repeat_writers.py:22]: Also, here is some relevant documentation that might prove helpful. https://www.postgresql.org/docs/current/queries-with.html

File: crt_portal/cts_forms/management/commands/flag_repeat_writers.py (hunk 3 above)
Comment (eileen-nava) [reply crt_portal/cts_forms/management/commands/flag_repeat_writers.py:34]: Sure, we can review during pairing.
Here's more relevant documentation. https://www.postgresql.org/docs/current/functions-matching.html
Here is a SO post that I appreciated. https://stackoverflow.com/questions/13287274/regexp-matches-better-way-to-get-rid-of-returning-curly-brackets

**Review by JackRyan1989 — APPROVED** [review]
Looks good to me! Testing worked on my local machine.

**Pull request merged**